### PR TITLE
bat: init at 0.2.0

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/bat/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  cargoSha256 = "07r10wwxv8svrw94djl092zj512868izlxldsiljfj34230abl02";
+
+  buildInputs = with pkgs; [ openssl pkgconfig cmake zlib file perl curl ];
+
+  meta = with stdenv.lib; {
+    description = "A cat(1) clone with syntax highlighting and Git integration";
+    homepage    = https://github.com/sharkdp/bat;
+    license     = licenses.mit;
+    maintainers = [];
```
Comment (xeji) [inline pkgs/tools/misc/bat/default.nix]: Would you like to be a maintainer of this package?

File: pkgs/tools/misc/bat/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    platforms   = platforms.unix;
```
Comment (xeji) [inline pkgs/tools/misc/bat/default.nix]: Please fix the build for Darwin or exclude it here.

**Comment by xeji** [comment]
```
installing
cp: -r not specified; omitting directory 'target/release'
cp: -r not specified; omitting directory 'target/release/deps'
cp: -r not specified; omitting directory 'target/release/.fingerprint'
cp: -r not specified; omitting directory 'target/release/native'
cp: -r not specified; omitting directory 'target/release/build'
cp: -r not specified; omitting directory 'target/release/examples'
cp: -r not specified; omitting directory 'target/release/incremental'
```
Is it safe to ignore these errors?

**Comment by matthiasbeyer** [comment]
I ran it successfully. I saw these errors, too. So I guess they can be ignored.

File: pkgs/tools/misc/bat/default.nix (hunk 1 above)
Comment (matthiasbeyer) [reply pkgs/tools/misc/bat/default.nix]: No, for reasons explained in https://github.com/NixOS/nixpkgs/pull/38504

File: pkgs/tools/misc/bat/default.nix (hunk 2 above)
Comment (matthiasbeyer) [reply pkgs/tools/misc/bat/default.nix]: About to do this.

**Pull request merged**